### ISSUE TITLE
Rule: don't override CSS styling outside plugin

### ISSUE THREAD

**Issue by saberzero1** [obsidianmd/eslint-plugin#24]
ESLint 9 supports CSS: https://eslint.org/blog/2025/02/eslint-css-support/

Not sure how to approach this yet. Perhaps check on overriding Obsidian's CSS custom properties? Perhaps start with checking in specific parts (e.g. `:root`, `body`, etc.)

**Comment by saberzero1** [comment]
> What's the thinking behind this?

Basically, a plugin should only target itself with CSS styling, not things outside itself. (Unless that is the explicit goal of the plugin, such as the Style Settings plugin)

For example, if you create a plugin that has a `button` with some custom styling, that button styling should not affect buttons outside your plugin.

```css
/* good, targets the plugin's button */
button.myPlugin {
  color: red;
}

/* bad, target too generic */
button {
  color: red;
}
```
